cmd/oras/internal/option/pusher.go

```diff
@@ -73,3 +98,22 @@ func decodeJSON(filename string, v interface{}) error {
 	defer file.Close()
 	return json.NewDecoder(file).Decode(v)
 }
+
+// getAnnotationsMap resharps annotationslice to k-v type and updates annotations
+func getAnnotationsMap(ManifestAnnotations []string, annotations map[string]map[string]string) error {
+	re := regexp.MustCompile(`=\s*`)
+	rawAnnotationsMap := make(map[string]string)
+	for _, rawAnnotationStr := range ManifestAnnotations {
+		rawAnnotation := re.Split(rawAnnotationStr, 2)
+		if annotationLength := len(rawAnnotation); annotationLength != 2 {
+			return errAnnotationFormat
+		}
+		rawAnnotation[0], rawAnnotation[1] = strings.TrimSpace(rawAnnotation[0]), strings.TrimSpace(rawAnnotation[1])
+		if _, ok := rawAnnotationsMap[rawAnnotation[0]]; ok {
+			return errAnnotationDuplication
```

Comment:
you also need tell use which key is duplicated via the `errAnnotationDuplication`, suggest to use `fmt.Errorf` and `%w` to wrap the error type.

Reply:
updated